src/redis.c

```diff
@@ -353,6 +353,41 @@ static int redis_handle_query(redisContext *rh, redis_node_t *rn,
   return 0;
 } /* }}} int redis_handle_query */
 
+static int redis_db_stats(char *node, char const *info_line) /* {{{ */
+{
+  unsigned char db;
+  unsigned char max_db = 16;
+  static char field_name[10];
+  static char db_id[3];
+  char *str;
+  static char buf[MAX_REDIS_VAL_SIZE];
+  value_t val;
+  int i;
+
+  for (db = 0; db < max_db; db++) {
+	ssnprintf(field_name, sizeof(field_name), "db%d:keys", db);
```

Comment:
The `%d` format expects an `int` but `db` is actually an `unsigned char`, which has the potential for an out of bounds read. As mentioned above, simply change `db` to an `int` to fix this issue.